[PowerRename]Add `$`, `^` and quantifiers to RegEx cheatsheet

PesBandi opened the pull request:
## Summary of the Pull Request
Adds these hints to the Regex cheatsheet:
* `^`: Start of the string
* `$`: End of the string
* `+`: Matches one or more of the preceding token
* `?`: Matches zero or one of the preceding token
* `*`: Matches zero or more of the preceding token

Also updates the `\S` description to remove an unnecessary backslash (resw doesn't need escapes)
## PR Checklist
- [x] **Closes:** #35847
- [x] **Communication:** I've discussed this with core contributors already.
- [x] **Tests:** All pass
- [x] **Localization:** All end user facing strings can be localized
- [x] **Dev docs:** No need
- [x] **New binaries:** None
- [x] **Documentation updated:** No need
## Detailed Description of the Pull Request / Additional comments
(The typo is fixed)
![image](https://github.com/user-attachments/assets/bb068cfa-0458-4f92-bf9c-a9485ee2db02)
## Validation Steps Performed
Manually tested

crutkas commented:
/azp run

moooyo commented:
Hi @PesBandi 

Currently, we have '.' in the cheatsheet but do not include '*' and '?'. I think we don't need to introduce the same pattern twice.

It might be better to add '.' and '?' to the cheatsheet and use a single pattern string (such as '\S+', '.*', or '\d+') to show this usage.

In most case of powerRename, we don't need '^' and '$' to indicate the start and end. So, I think it's no need to add them. If someone need, they can check the help link.

PesBandi commented:
Hi @moooyo, thanks for the review. I totally agree with you! Now that I'm looking at the cheatsheet I see that it could use some changes. Maybe add descriptions for `*` and `?` as you suggested and remove `\S+`, since I believe that the user can figure that out? I really don't have a strong opinion here, so if you have any specific ideas it would really help. Also cc @htcfreek

htcfreek commented:
@PesBandi , @moooyo 
Let's have `*` and `+` as single entries after the `.` and remove the combination entries.

stefansjfw commented:
> @PesBandi , @moooyo Let's have `*` and `+` as single entries after the `.` and remove the combination entries.

Hi @PesBandi, are you going to update the PR according to this comment?

PesBandi commented:
Hi @stefansjfw, of course! I was planning to do it yesterday, but I haven't had the time yet. You can expect it to be done today.

PesBandi commented:
Hi @stefansjfw, @htcfreek
I updated it according to htcfreek's comment, you can review it now. Thanks
<details><summary>Screenshot</summary>

![image](https://github.com/user-attachments/assets/c59e55d6-40e8-4ff7-bc8c-32cf97d5e887)
</details>

htcfreek commented:
@PesBandi 
We need `?` (zero or one times) too. Sory for missing to mention this.

PesBandi commented:
@htcfreek, thanks for the review!
1. I forgot about that one, it's added now.
2. Didn't notice that, added it back.
3. Yes it does exist. I don't think we should add it. IMO it's not used as frequently and the list is already really long. We can't explain all of RegEx in a single flyout and I believe that `\W` should be sacrificed to make it shorter. Furthermore, the user can figure out what the uppercase version of an escape does from `\s` and `\S`.
4. Yes, there is a link to https://aka.ms/powertoysRegExHelp
5. `\w` matches a word character. `\b` matches a word boundary i.e. a character next to a word character. The order doesn't matter, but I updated it so that it's the same for both.

<details><summary>Screenshot</summary>

![image](https://github.com/user-attachments/assets/bb068cfa-0458-4f92-bf9c-a9485ee2db02)
</details>

htcfreek commented:
@PesBandi 
Shouldn't it be `a non-digit character`?

PesBandi commented:
@htcfreek, which one exactly? I am not sure what you are talking about.

htcfreek commented:
> @htcfreek, which one exactly? I am not sure what you are talking about.

\D

PesBandi commented:
Oh, you mean adding the word *character*, right? I haven't changed that one, but I agree with you, I'll add it.

PesBandi commented:
On second thought though, the description for `\d` is `Any digit, short for [0-9]`, which in my opinion sounds better than `Any digit character, short for [0-9]`. Right now `\D` is consistent with `\d`. Adding *character* would break this consistency and only eliminate a questionable amount of ambiguity. IMHO it's not worth it, but I don't know what your style guide is.

moooyo commented:
Emmm.  Anyone can summarize our current status? Which one will be listed in the pop up box?

I agree with @PesBandi's opnion. We can not list all of the usage in this popup box. So, we need to chose the most important or frequently one. If we have a frequency table, that's perfect. (but maybe we don't have I believe).

From my view, I think '^', '$', '.', '?', '+', '*' is important. And '\d' is also frequent to use.  For white space, maybe we can use ' ' directly? Why not? I think no one will add '\r\n' into the file name.

PesBandi commented:
@moooyo the screenshot in the PR description is the current state.
`\s` is not just `[ \n\r]`, it's `[\f\n\r\t\v\u0020\u00a0\u1680\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]`. It is true though that most of these are really niche characters and maybe we should remove it in favor of something more useful.

stefansjfw commented:
/s can be useful for removing whitespaces from filenames, so let's keep it
